### ISSUE TITLE
Regression:  #9122 breaks Grid init

### ISSUE THREAD

**Issue by manolo** [vaadin/flow#9136]
vaadin-flow-components master build is failing after flow-5.0-SNAPSHOT was updated.
git bisect reported that the culprit is a fix for disabled attribute (28115e1392131866eb53323959677d96c53055bb)

How to reproduce:
```
$ git clone git@github.com:vaadin/vaadin-flow-components.git
$ cd vaadin-flow-components
$ mvn -am -pl vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests verify -Dit.test=GridWithTemplateIT 
```

Test fails with the following output
```
Caused by: java.lang.NullPointerException
	at com.vaadin.flow.component.grid.Grid.onEnabledStateChanged(Grid.java:3305)
	at com.vaadin.flow.dom.Element.informEnabledStateChange(Element.java:1587)
	at com.vaadin.flow.dom.Element.setEnabled(Element.java:1561)
	at com.vaadin.flow.component.Component.mapToElement(Component.java:197)
	at com.vaadin.flow.component.Component.<init>(Component.java:118)
	at com.vaadin.flow.component.grid.Grid.<init>(Grid.java:1322)
	at com.vaadin.flow.component.grid.Grid.<init>(Grid.java:1196)
	at com.vaadin.flow.component.grid.Grid.<init>(Grid.java:1181)
```

**Comment by manolo** [comment]
The problem seems to appear when the grid is instantiated using a template.
It could probably be a problem in the component not checking if the `dataComunnicator` exists, or not instantiating it before, but it needs further investigation in order not to break customer components.

**Comment by denis-anisimov** [comment]
This specific problem is in the `Grid` :  `dataCommunicator` should be checked for `null`  before calling it.

But yes, this an issue which has a lot of various aspects.
The main issue is : `Component::from` and it's delegate `ComponentUtil::componentFromElement` : it allows to wrap an existing `Element` to a `Component`.
As a result a `Component` is created with already existing `Element` and it's quite unexpected: normally when you instantiate a component you expect that `Element`  is not yet available and normally no-one checks `Element` (via API) properties/attributes.
(as a side note : one proper step could be making a CTOR which accepts an `Element`  as a parameter but we may not force it since it's possible to have a default CTOR which may be called with an existing `Element`).

This is already a known issue which has been fixed by me in many our `Component`  which extend `AbstractSinglePropertyField` : every such component is not aware of the fact that some property may be already set in the underlying `Element` (e.g. `"value"`)   and just set it explicitly in the CTOR just because it's not possible to know that you have to care about it.
The fix has been done in our components only but other components may be affected.

Another aspect is `setEnabled`  contract : it behaves differently when `Element`  is not mapped to a `Component`  and when it's mapped. When there is no a component the `"disabled"`  attribute is not set ( there are also some calls for children besides  the `"disabled"`  attribute set for a component).  But when it's mapped then a `Component`  method is executed which set the `"disabled"`  attribute and do some calls for children.
As a result this method:
- may be overridden
- if `Element::setEnabled`  is called in the `Component` CTOR (as it's done in the fix #9122) it will call the method which may be overridden
- the overriden method is not aware of the fact that it's called from the superclass and this is a classical issue: some fields in the class which are initialized in the CTOR (or at the declaration) are not yet available and this is absolutely unexpected

All those things are consequences of the bad and unclear design of `Component::from` / component mapping (which is used every time on `@Id` injected fields).
This design is the root cause of all the issues and it requires review and rewriting. Otherwise it allows to introduce bugs which you are totally unaware of at every level : inside the core itself and outside it, in the user implementation because there is no simply any contract to get any knowledge that you have to care about some specific situations.
